### PR TITLE
deploy: set gracePeriod on deployer creation rather than when deleting

### PR DESCRIPTION
**Pull request by kargakis**
@smarterclayton

### CODE REVIEW

File: pkg/deploy/controller/deployment/controller.go [hunk 1]
```diff
@@ -275,6 +275,8 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 	// Assigning to a variable since its address is required
 	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds
 
+	gracePeriod := int64(10)
```
Comment (smarterclayton) [inline pkg/deploy/controller/deployment/controller.go]: Does the deployer have to wait for child pod hooks to finish?

Comment (kargakis) [inline pkg/deploy/controller/deployment/controller.go]: On deletion I wouldn't think so.

Comment (smarterclayton) [inline pkg/deploy/controller/deployment/controller.go]: So the controllers will delete those hook pods?

Comment (kargakis) [inline pkg/deploy/controller/deployment/controller.go]: Yes, the deployment controller is deleting all hooks and the deployer for the deployment it is reconciling:
https://github.com/openshift/origin/blob/26012f2fe41a6518f6ffd97c8b780c6d2c9c59e4/pkg/deploy/controller/deployment/controller.go#L365
We also test this in unit tests, eg:
https://github.com/openshift/origin/blob/26012f2fe41a6518f6ffd97c8b780c6d2c9c59e4/pkg/deploy/controller/deployment/controller_test.go#L466

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**